astroplan/constraints.py

```diff
@@ -0,0 +1,565 @@
+    """
+    Are the ``targets`` always observable throughout ``time_range`` given
+    constraints in ``constraints_list`` for ``observer``?
+
+    Parameters
+    ----------
+    constraints : list or `~astroplan.constraints.Constraint`
+        Observational constraint(s)
+
+    time_range : `~astropy.time.Time`
+            Lower and upper time bounds on which to compute the constraints.
+            If input is a scalar time, will only compute at that time (no
+            range).
+
+    targets : {list, `~astropy.coordinates.SkyCoord`, `~astroplan.FixedTarget`}
+        Target or list of targets
+
+    observer : `~astroplan.Observer`
+        The observer who has constraints ``constraints``
+
+    time_resolution : `~astropy.units.Quantity` (optional)
+        Determine whether constraints are met between test times in
+        ``time_range`` by checking constraint at linearly-spaced times separated
+        by ``time_resolution``. Default is 0.5 hours.
+
+    Returns
+    -------
+    ever_observable : list
+        List of booleans of same length as ``targets`` for whether or not each
+        target is observable in the time range given the constraints.
+    """
+    if not hasattr(constraints, '__len__'):
+        constraints = [constraints]
+
+    contraint_arr = np.logical_and.reduce([constraint(time_range, observer,
+                                                      targets,
+                                                      time_resolution=time_resolution)
+                                          for constraint in constraints])
+    return np.all(contraint_arr, axis=1)
+
+def is_observable(constraints, time_range, targets, observer,
+                  time_resolution=DEFAULT_TIME_RESOLUTION):
+    """
+    Are the ``targets`` observable during ``time_range`` given constraints in
+    ``constraints_list`` for ``observer``?
+
+    Parameters
+    ----------
+    constraints : list or `~astroplan.constraints.Constraint`
+        Observational constraint(s)
+
+    time_range : `~astropy.time.Time`
+        Lower and upper time bounds on which to compute the constraints.
+        If input is a scalar time, will only compute at that time (no
+        range).
+
+    targets : {list, `~astropy.coordinates.SkyCoord`, `~astroplan.FixedTarget`}
+        Target or list of targets
+
+    observer : `~astroplan.Observer`
+        The observer who has constraints ``constraints``
+
+    time_resolution : `~astropy.units.Quantity` (optional)
+        Determine whether constraints are met between test times in
+        ``time_range`` by checking constraint at linearly-spaced times separated
+        by ``time_resolution``. Default is 0.5 hours.
+
+    Returns
+    -------
+    ever_observable : list
+        List of booleans of same length as ``targets`` for whether or not each
+        target is ever observable in the time range given the constraints.
+    """
+    if not hasattr(constraints, '__len__'):
```

Comment:
can you clarify why you're using `hasattr` here instead of, say `astropy.utils.isiterable`? Not sure it really matters much, but thought I'd check